### ISSUE TITLE
Feature request: Kinesis Data Firehose server-side encryption

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#6443]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Server-side encryption for Amazon Kinesis Data Firehose [recently announced](https://docs.aws.amazon.com/firehose/latest/dev/encryption.html).
Requires:
* https://github.com/terraform-providers/terraform-provider-aws/pull/6439

**Comment by ewbankkit** [comment]
It seems that Kinesis Firehose delivery stream tagging is also enabled with this SDK release.

**Comment by ewbankkit** [comment]
Proposed syntax:

```hcl
resource "aws_kinesis_firehose_delivery_stream" "test_stream" {
  name                   = "terraform-kinesis-firehose-test-stream"
  destination            = "s3"
  server_side_encryption = true

  ...
}
```

`server_side_encryption ` will be ignored if `kinesis_source_configuration` is specified as it's the Kinesis data stream's [responsibility](https://docs.aws.amazon.com/firehose/latest/dev/encryption.html#sse-with-data-stream-as-source) to [enable](https://www.terraform.io/docs/providers/aws/r/kinesis_stream.html#encryption_type) server-side encryption.

**Comment by bflad** [comment]
Support for the new `server_side_encryption` configuration block has been merged and will release with version 2.27.0 of the Terraform AWS Provider, in about a week. 👍 Thanks to @ewbankkit for the implementation.
